luxonis_ml/data/augmentations/augmentations.py

```diff
@@ -0,0 +1,359 @@
+        else:
+            resize = A.Resize(height=height, width=width)
+
+        if is_validation_pipeline:
+            config = [a for a in config if a.get("name") == "Normalize"]
+
+        pixel_augs = []
+        spatial_augs = []
+        batched_augs = []
+        batch_size = 1
+        for aug in config:
+            curr_aug = cls._get_augmentation(
+                aug["name"], **aug.get("params", {})
+            )
+            if isinstance(curr_aug, A.ImageOnlyTransform):
+                pixel_augs.append(curr_aug)
+            elif isinstance(curr_aug, BatchBasedTransform):
+                batch_size *= curr_aug.batch_size
+                batched_augs.append(curr_aug)
+            elif isinstance(curr_aug, A.DualTransform):
+                spatial_augs.append(curr_aug)
+
+        def _get_params():
+            return {
+                "bbox_params": A.BboxParams(
+                    format="albumentations",
+                    label_fields=["bboxes_classes", "bboxes_visibility"],
+                    # Bug in albumentations v1.4.18 (the latest installable
+                    # on python 3.8) causes the pipeline to eventually
+                    # crash when set to 0.
+                    min_visibility=0.01,
+                ),
+                "keypoint_params": A.KeypointParams(
+                    format="xy",
+                    label_fields=["keypoints_visibility", "keypoints_classes"],
+                    remove_invisible=False,
+                ),
+            }
+
+        return cls(
+            height=height,
+            width=width,
+            batch_size=batch_size,
+            batch_transform=BatchCompose(batched_augs, **_get_params()),
+            spatial_transform=A.Compose(spatial_augs, **_get_params()),
+            pixel_transform=A.Compose(pixel_augs),
+            resize_transform=A.Compose([resize], **_get_params()),
+        )
+
+    @property
+    @override
+    def batch_size(self) -> int:
+        return self._batch_size
+
+    @override
+    def apply(self, data: List[LoaderOutput]) -> LoaderOutput:
+        random_state = random.getstate()
+        np_random_state = np.random.get_state()
+
+        task_names = {
+            get_qualified_task_name(task)
+            for _, label in data
+            for task in label
+        }
+        out_labels = {}
+
+        if not task_names:
+            return self._apply(data, 0, 0)
+
+        for task_name in task_names:
+            task_types_to_names = {}
+            batch_data: List[LoaderOutput] = []
+            nk = 0
+            ns = 0
+            for img, labels in data:
+                label_subset = {}
+                for task, label in labels.items():
+                    if get_qualified_task_name(task) == task_name:
+                        task_type = get_task_type(task)
+                        if task_type == "instance_segmentation":
+                            task_type = "segmentation"
```

Comment:
Why the task_type is changed here?

Reply:
From albumentation's perspective instance segmentation is treated the same as normal segmentation.